perllib/FixMyStreet/Queue/Item/Report.pm

```diff
@@ -202,27 +225,55 @@ sub _create_reporters {
 
     if (FixMyStreet->staging_flag('send_reports', 0)) {
         # on a staging server send emails to ourselves rather than the bodies
-        %reporters = map { $_ => $reporters{$_} } grep { /FixMyStreet::SendReport::Email/ } keys %reporters;
-        unless (%reporters) {
-            %reporters = ( 'FixMyStreet::SendReport::Email' => FixMyStreet::SendReport::Email->new() );
+        unless (%email_reporters) {
+            %email_reporters
+                = ( EMAIL_SENDER_PREFIX => ${ \EMAIL_SENDER_PREFIX }->new() );
         }
+
+        @other_reporters = ();
     }
 
-    $self->_set_reporters(\%reporters);
+    $self->_set_reporters([@other_reporters, values %email_reporters]);
 }
 
 sub _send {
     my $self = shift;
 
     # Multiply results together, so one success counts as a success.
     my $result = -1;
+    my $report = $self->report;
+
+    my @add_send_fail_body_ids;
+    my @remove_send_fail_body_ids;
+
+    for my $sender ( @{ $self->reporters } ) {
+        my $sender_name = ref $sender;
+        $self->log( 'Sending using ' . $sender_name );
+
+        if ( FixMyStreet->staging_flag( 'send_reports', 0 ) ) {
+            # If on staging with send_reports set to 0, bypass the body_id
+            # logic below, as we may only have a special Email reporter with
+            # no bodies attached
```

Comment:
Sorry, now I've seen this with the most recent changes, I can also see that I don't think this block is needed either now :) The `else` block will work fine regardless of the `send_reports` setting - in the case there's a special Email reporter as per this comment, @body_ids is empty and it all works as you'd expect.